Source/Core/UICommon/UICommon.cpp

```diff
@@ -295,7 +295,22 @@ void SetUserDirectory(std::string custom_path)
   //    -> Use AppData\Roaming\Dolphin Emulator as the User directory path
   // 6. Default
   //    -> Use GetExeDirectory()\User
+  //
+  // On Steam builds, we take a simplified approach:
+  // 1. GetExeDirectory()\portable.txt exists
+  //    -> Use GetExeDirectory()\User
+  // 2. AppData\Roaming exists
+  //    -> Use AppData\Roaming\Dolphin Emulator (Steam) as the User directory path
```

Comment:
I would actually suggest the Steam builds not use AppData\Roaming at _**all**_, and instead rely on Steam user-specific folders. Using folders outside the Steam directory could cause conflicts if multiple Steam users sign into the same OS user. (Sorry I didn't think to say this sooner...)